pylint/config/config_initialization.py

```diff
@@ -18,18 +18,23 @@ def _config_initialization(
     args_list: List[str],
     reporter: Union[reporters.BaseReporter, reporters.MultiReporter, None] = None,
     config_file: Union[None, str, Path] = None,
-    verbose_mode: Optional[bool] = None,
+    verbose_mode: bool = False,
 ) -> List[str]:
     """Parse all available options, read config files and command line arguments and
     set options accordingly.
     """
+    # Convert config_file in a Path and string
+    config_file_path = Path(config_file) if config_file else None
+    config_file = str(config_file) if config_file else None
```

Comment:
Why do we need both ?  
```suggestion
    config_file = Path(config_file) if config_file else None
```
The we cast to string when/if required ?

Reply:
I opted to do it like this to give some flexibility in later PRs and keep those diffs smaller.

Reply:
All right !